### PR TITLE
Support dingtalk reporter

### PR DESCRIPTION
**Pull request by sawyer523**
Here are the steps to send Prow job status notifications to a DingTalk chat group, similar to notifications in a Slack channel:
  1. Create a custom chatbot in the DingTalk group.
  1. Note down the Webhook URL for the chatbot.
  1. Define a DingTalk alerter in the Prow config file (e.g. config.yaml):
  ```yaml
  dingtalk_reporter_configs:
  '*':
    job_types_to_report:
      - postsubmit
      - presubmit
    job_states_to_report:
      - success
      - failure
      - error
    token: < The token_secret value in the webhook.>
    report_template: `{{ $repo := "" }}{{with .Spec.Refs}}{{$repo = .Repo}}{{end}}{{if eq $repo ""}}{{if .Spec.ExtraRefs}}{{with index .Spec.ExtraRefs 0}}{{$repo = .Repo}}{{end}}{{end}}{{end}}## Repo: {{ $repo }}
---
- Job: {{.Spec.Job}}
- Type: {{.Spec.Type}}
- State: {{if eq .Status.State "triggered"}}<font color="orange">**{{.Status.State}}**</font>{{end}}{{if eq .Status.State "pending"}}<font color="yellow">**{{.Status.State}}**</font>{{end}}{{if eq .Status.State "success"}}<font color="green">**{{.Status.State}}**</font>{{end}}{{if eq .Status.State "failure"}}<font color="red">**{{.Status.State}}**</font>{{end}}{{if eq .Status.State "aborted"}}<font color="gray">**{{.Status.State}}**</font>{{end}}{{if eq .Status.State "error"}}<font color="red">**{{.Status.State}}**</font>{{end}}
- Log: [View logs]({{.Status.URL}})`
  ```

The received notification is similar to this:
image

### CODE REVIEW

**Comment by bart0sh** [comment]
@sawyer523 Thank you for your PR. Please [sign the CLA](https://github.com/kubernetes/community/tree/master/contributors/guide/contributor-cheatsheet#signing-the-cla) to proceed further, thanks.

Can you also update PR description so reviewers would understand better what this change is for?

**Comment by sawyer523** [comment]
> @sawyer523 Thank you for your PR. Please [sign the CLA](https://github.com/kubernetes/community/tree/master/contributors/guide/contributor-cheatsheet#signing-the-cla) to proceed further, thanks.
> 
> Can you also update PR description so reviewers would understand better what this change is for?

I have signed the CLA as you requested.

I apologize that my initial pull request description was unclear. Per your suggestion, I have updated it with more details on the purpose 翻译一下

File: jobs/e2e_node/crio/README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 [@kubernetes/sig-node-cri-o-test-maintainers](https://github.com/orgs/kubernetes/teams/sig-node-cri-o-test-maintainers)
 on the related issue or pull request.**
 
-All jobs maintained within this directoy are part of the `sig-node-cri-o`
+All jobs maintained within this directory are part of the `sig-node-cri-o`
```
Comment (SergeyKanzhelev) [inline jobs/e2e_node/crio/README.md:7]: this is the only reason it is marked for sig-node. consider splitting it to the different PR.

File: prow/dingtalk/client.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+		return fmt.Errorf("failed to post message to %s: %w", token, err)
+	}
+	return nil
+}
+
+//`{
+//    "msgtype": "markdown",
+//    "markdown": {
+//        "title": "notify",
+//        "text": "## Job {{.Spec.Job}}\n---\n- Type: {{.Spec.Type}}\n- State: {{if eq .Status.State "triggered"}}<font color="orange">**{{.Status.State}}**</font>{{end}}{{if eq .Status.State "pending"}}<font color="yellow">**{{.Status.State}}**</font>{{end}}{{if eq .Status.State "success"}}<font color="green">**{{.Status.State}}**</font>{{end}}{{if eq .Status.State "failure"}}<font color="red">**{{.Status.State}}**</font>{{end}}{{if eq .Status.State "aborted"}}<font color="gray">**{{.Status.State}}**</font>{{end}}{{if eq .Status.State "error"}}<font color="red">**{{.Status.State}}**</font>{{end}}\n- Log: [View logs]({{.Status.URL}})"
```
Comment (liangyuanpeng) [inline prow/dingtalk/client.go:128]: This  is not necessary?

Comment (sawyer523) [reply prow/dingtalk/client.go]: Yes, I have already deleted this code.

File: jobs/e2e_node/crio/README.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # CRI-O test infra jobs
 
-**For any modifications to this directory, please ping
+**For any modifications to this directoy, please ping
```
Comment (jihoon-seo) [inline jobs/e2e_node/crio/README.md:3]:
```suggestion
**For any modifications to this directory, please ping
```

**Review by jihoon-seo — COMMENTED** [review]
Hello @sawyer523, the Prow source code has been moved to https://github.com/kubernetes-sigs/prow on April 9, 2024.
Please consider migrate this PR (by opening a new one in there).
Thanks!

/remove-area config prow/bump
/remove-sig node testing

**Pull request closed without merging**